apps/web/src/services/dashboards/widgets/compliance-check-status/ComplianceCheckStatusWidget.vue

```diff
@@ -317,8 +321,20 @@ const drawChart = (outerChartData, innerChartData) => {
 
 const initWidget = async (data?: Data): Promise<Data> => {
     state.loading = true;
-    state.data = data ?? await fetchData();
-    if (!data) state.severityData = await fetchSeverityData();
+    if (data) {
+        state.complianceData = data.compliance;
+        state.severityData = data.severity;
+    } else {
+        Promise.all([
+            fetchComplianceData(),
+            fetchSeverityData(),
+        ]).then(([complianceData, severityData]) => {
+            state.data = {
+                compliance: complianceData,
+                severity: severityData,
+            };
+        });
```

Comment:
suggestion: what about to use async await instead of using callback? :)

```suggestion
        const [complianceData, severityData] = await Promise.all([
            fetchComplianceData(),
            fetchSeverityData(),
        ])
        state.data = {
            compliance: complianceData,
            severity: severityData,
        };
```

Reply:
wow thanks